backend/apps/ecommerce/tests.py

```diff
@@ -0,0 +1,220 @@
+
+    def capture_payment(order, method):
+        print("Order: ", order, "\nMethod: ", method)
+
+
+class EcommerceMutationsTestCase(EcommerceBaseTestCase):
+    """
+    Testing all mutations for ecommerce
+    """
+
+    def test_create_product(self):
+        """
+        THIS TEST MUST BE REWRITTEN!
+        """
+
+        query = """
+          mutation {
+            createProduct {
+                product {
+                    id
+                    name
+                    description
+                    price
+                    totalQuantity
+                    maxBuyableQuantity
+                }
+                ok
+                }
+            }
+        """
+        response = self.query(query)
+        self.assertResponseNoErrors(response)
+
+        data = json.loads(response.content)["data"]
+        response_product = data["createProduct"]["product"]
+        product = Product.objects.get(pk=response_product["id"])
+        if product is None:
+            self.assertIsNotNone(product, msg="Expected product after creation, got None")
+        else:
+            self.assertTrue(data["createProduct"]["ok"])
+            product.price = int(product.price)
+            response_product["price"] = int(response_product["price"])
+            self.deep_assert_equal(response_product, product)
+
+    @unittest.mock.patch.object(InitiateOrder, "vipps_api", VippsApiMock)
+    def test_initiate_order(self):
+        query = (
+            lambda q: f"""
+        mutation InitiateOrder {{
+            initiateOrder(productId: {self.product_1.id}, quantity: {q}) {{
+                redirect
+            }}
+        }}
+        """
+        )
+
+        # Unauthorized user should not be able to initiate order
+        response = self.query(query(1))
+        self.assert_permission_error(response)
+
+        # Requesting more than available is not allowed
+        response = self.query(query(6), user=self.indok_user)
+        self.assertResponseHasErrors(response)
+
+        # Requesting more than one user can buy is not allowed
+        response = self.query(query(3), user=self.indok_user)
+        self.assertResponseHasErrors(response)
+
+        # Max is 2 per user so should be allowed to buy this
+        response = self.query(query(2), user=self.indok_user)
+        self.assertResponseNoErrors(response)
+        data = json.loads(response.content)["data"]
+        redirect_url = data["initiateOrder"]["redirect"]
+        self.assertEqual(redirect_url, "https://www.youtube.com/watch?v=dQw4w9WgXcQ")
+
+        # Already bought the total quota for the user so shouldn't be allowed to buy more
+        # Need to think a bit more on this one as only captured orders are counted with this
+        # response = self.query(query(1), user=self.indok_user)
+        # self.assertResponseHasErrors(response)
+
+    @unittest.mock.patch.object(AttemptCapturePayment, "vipps_api", VippsApiMock)
+    def test_attempt_capture_order(self):
```

Comment:
Are we also testing the happy path here? I.e. where a user successfully captures a payment